docs/canonicalk8s/snap/howto/intermediate-ca.md

```diff
@@ -0,0 +1,60 @@
+# How to use intermediate CAs with Vault
+
+By default, {{product}} will generate self-signed CA certificates
+for the Kubernetes services.
+
+Follow this guide to prepare an intermediate Certificate Authority (CA) using
+[HashiCorp Vault] and then configure {{product}} to use the generated
+certificates.
+
+```{include} /_parts/common_vault_intermediate_ca.md
+```
+
+## Pass intermediate CA certificates to {{product}}
+
+The CA certificates can be specified through the [bootstrap configuration file]
+using the following fields:
+
+| Field                  | Description                |
+|------------------------|----------------------------|
+| ``ca-crt``             | API server CA certificate  |
+| ``ca-key``             | API server CA key          |
+| ``client-ca-crt``      | client CA certificate      |
+| ``client-ca-key``      | client CA key              |
+| ``front-proxy-ca-crt`` | Front Proxy CA certificate |
+| ``front-proxy-ca-key`` | Front Proxy CA key         |
+
+Let's prepare a bootstrap configuration using our newly generated intermediate
+CA certificate.
+
+```
+cat <<EOF > myca/bootstrap.yaml
+ca-crt: |
+$(cat myca/intermediate.crt | sed 's/^/    /g')
+ca-key: |
+$(cat myca/intermediate.key | sed 's/^/    /g')
+cluster-config:
+  network:
+    enabled: true
+  dns:
+    enabled: true
+  local-storage:
+    enabled: true
+EOF
+```
+
+Now bootstrap the cluster:
+
+```
+k8s bootstrap --file myca/bootstrap.yaml
+```
+
+# Further reading
+
+See this [Vault article] for more details on how to integrate Vault as a
+Kubernetes certificate manager.
+
+<!--LINKS -->
+[HashiCorp Vault]: https://developer.hashicorp.com/vault/docs
+[bootstrap configuration file]: /snap/reference/bootstrap-config-reference.md
```

Comment:
This link is out of date. new location /snap/reference/config-files/bootstrap-config.md

Reply:
Got it, we'll have to update this link as well: https://github.com/canonical/k8s-snap/blob/8fda72f4671a423e70458b15d3602fe47c5cffe0/docs/canonicalk8s/snap/howto/install/custom-bootstrap-config.md?plain=1#L83